packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m

```diff
@@ -87,13 +88,25 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
     [[GIDSignIn sharedInstance] signOut];
     result(nil);
   } else if ([call.method isEqualToString:@"disconnect"]) {
-    [_accountRequests insertObject:result atIndex:0];
-    [[GIDSignIn sharedInstance] disconnect];
+    if ([self setAccountRequest:result]) {
+      [[GIDSignIn sharedInstance] disconnect];
+    }
   } else {
     result(FlutterMethodNotImplemented);
   }
 }
 
+- (BOOL)setAccountRequest:(FlutterResult)request {
+  if (_accountRequest != nil) {
+    request([FlutterError errorWithCode:@"concurrent-requests"
```

Comment:
Agreed. Makes me wonder whether we should be catching _all_ exceptions in the plugin handler itself and converting them to errors.

Rationale is that a plugin should not cause an app crash in native layer. All errors should be reported straight to the app. That way, if you have a misbehaving plugin, the app at least has some control over what to do about it. Wdyt?